src/main/java/cloud/eppo/EppoClient.java

```diff
@@ -412,9 +409,67 @@ private JsonNode parseJsonString(String jsonString) {
     }
   }
 
+  public BanditResult getBanditAction(
+      String flagKey,
+      String subjectKey,
+      DiscriminableAttributes subjectAttributes,
+      Actions actions,
+      String defaultValue) {
+    BanditResult result = new BanditResult(defaultValue, null);
+    try {
+      String assignedVariation =
+          getStringAssignment(
+              flagKey, subjectKey, subjectAttributes.getAllAttributes(), defaultValue);
+
+      // Update result to reflect that we've been assigned a variation
+      result = new BanditResult(assignedVariation, null);
+
+      String banditKey = configurationStore.banditKeyForVariation(flagKey, assignedVariation);
+      if (banditKey != null && !actions.isEmpty()) {
+        BanditParameters banditParameters = configurationStore.getBanditParameters(banditKey);
+        BanditEvaluationResult banditResult =
+            BanditEvaluator.evaluateBandit(
+                flagKey, subjectKey, subjectAttributes, actions, banditParameters.getModelData());
+
+        if (banditLogger != null) {
+          BanditAssignment banditAssignment =
+              new BanditAssignment(
+                  flagKey,
+                  banditKey,
+                  subjectKey,
+                  banditResult.getActionKey(),
+                  banditResult.getActionWeight(),
+                  banditResult.getOptimalityGap(),
+                  banditParameters.getModelVersion(),
+                  subjectAttributes.getNumericAttributes(),
+                  subjectAttributes.getCategoricalAttributes(),
+                  banditResult.getActionAttributes().getNumericAttributes(),
+                  banditResult.getActionAttributes().getCategoricalAttributes(),
+                  buildLogMetaData());
+
+          banditLogger.logBanditAssignment(banditAssignment);
+        }
+
+        // Update result to reflect that we've been assigned an action
+        result = new BanditResult(assignedVariation, banditResult.getActionKey());
```

Comment:
result should get set above the call to `banditLogger.logBanditAssignment`. This ensures that if the user-provided bandit logger throws an exception, the `throwIfNotGraceful` call below has an accurate result to return.

Reply:
Great point. That should be done _and_ all logger calls should be wrapped in try-catches so they are non fatal